### PR TITLE
Adding validation when constructing a Local module

### PR DESCRIPTION
**Pull request by zhangchiqing**
Adding a missing validation to prevent from using a mismatching private key

### CODE REVIEW

File: module/local/me.go [hunk 1]
```diff
@@ -15,6 +17,11 @@ type Local struct {
 }
 
 func New(id *flow.Identity, sk crypto.PrivateKey) (*Local, error) {
+	if !sk.PublicKey().Equals(id.StakingPubKey) {
```
Comment (tarakby) [inline module/local/me.go:20]: 👌🏼

**Review by tarakby — APPROVED** [review]
🏄🏼

File: module/local/me_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestInitializeWithMatchingKey(t *testing.T) {
```
Comment (durkmurder) [inline module/local/me_test.go:11]: I would like to see at least a minimal godoc for each test. Just a quick summary.

**Review by durkmurder — APPROVED** [review]
:guitar:

**Comment by zhangchiqing** [comment]
bors merge

**Pull request merged**